src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/FloatingPointMutatorFactory.java

```diff
@@ -0,0 +1,164 @@
+    if (!(type.getType() instanceof Class)) {
+      return Optional.empty();
+    }
+    Class<?> clazz = (Class<?>) type.getType();
+
+    if (clazz == float.class || clazz == Float.class) {
+      return Optional.of(new AbstractFloatMutator<Float>(type, -Float.MAX_VALUE, Float.MAX_VALUE) {
+        @Override
+        protected Float mutateWithLibFuzzer(Float value) {
+          return LibFuzzerMutator.mutateDefault(value, this, 0);
+        }
+
+        @Override
+        public Float init(PseudoRandom prng) {
+          return (float) initImpl(prng);
+        }
+
+        @Override
+        public Float mutate(Float value, PseudoRandom prng) {
+          return (float) mutateImpl(value, prng);
+        }
+
+        @Override
+        public Float read(DataInputStream in) throws IOException {
+          float value = (float) in.readFloat();
+          return value;
+        }
+
+        @Override
+        public void write(Float value, DataOutputStream out) throws IOException {
+          out.writeFloat(value);
+        }
+      });
+    } else if (clazz == double.class || clazz == Double.class) {
+      return Optional.of(
+          new AbstractDoubleMutator<Double>(type, -Double.MAX_VALUE, Double.MAX_VALUE) {
+            @Override
+            protected Double mutateWithLibFuzzer(Double value) {
+              return LibFuzzerMutator.mutateDefault((Double) value, this, 0);
+            }
+
+            @Override
+            public Double init(PseudoRandom prng) {
+              return (double) initImpl(prng);
+            }
+
+            @Override
+            public Double mutate(Double value, PseudoRandom prng) {
+              value = (double) mutateImpl(value, prng);
+              return value;
+            }
+
+            @Override
+            public Double read(DataInputStream in) throws IOException {
+              double value = (double) in.readDouble();
+              return value;
+            }
+
+            @Override
+            public void write(Double value, DataOutputStream out) throws IOException {
+              out.writeDouble(value);
+            }
+          });
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  // Visible for testing.
+  static abstract class AbstractFloatMutator<T extends Float> extends SerializingMutator<T> {
+    AbstractFloatMutator(
+        AnnotatedType type, Float defaultMinValueForType, Float defaultMaxValueForType) {}
+
+    protected final Float initImpl(PseudoRandom prng) {
+      return mutateImpl(0.0f, prng);
+    }
+
+    protected final Float mutateImpl(Float value, PseudoRandom prng) {
+      return prng.closedRange(-Float.MAX_VALUE, Float.MAX_VALUE);
+    }
+
+    @ForOverride protected abstract Float mutateWithLibFuzzer(Float value);
+
+    @Override
+    public T detach(T value) {
+      // Always immutable.
+      return value;
+    }
+
+    @Override
+    public String toDebugString(Predicate<Debuggable> isInCycle) {
```

Comment:
If you override this method, you might as well just return `"Double"`. Also applies to the float mutator.

Reply:
Done!